src/coreclr/vm/jitinterface.cpp

```diff
@@ -7142,6 +7142,27 @@ bool getILIntrinsicImplementationForInterlocked(MethodDesc * ftn,
     return true;
 }
 
+bool IsBitwiseEquatable(TypeHandle typeHandle, MethodTable * methodTable)
+{
+    if (!methodTable->IsValueType() || methodTable->ContainsPointers())
+    {
+        return false;
+    }
+
+    if (!CanCompareBitsOrUseFastGetHashCode(methodTable))
+    {
+        return false;
+    }
+
+    Instantiation inst(&typeHandle, 1);
+    if (typeHandle.CanCastTo(TypeHandle(CoreLibBinder::GetClass(CLASS__IEQUATABLEGENERIC)).Instantiate(inst)))
```

Comment:
Is an overriden `Equals` in a struct sign that it should not be used for bitwise equality?

Reply:
Isn't that handled by https://github.com/dotnet/runtime/blob/53841055e735895520633ecf5da392ac14415ba8/src/coreclr/vm/comutilnative.cpp#L1670?

Reply:
@stephentoub thanks, it's interesting that that one doesn't take `IEquatable<T>` into account 🙂 so I assume fast hashcode/fast equals on structs might ignore it

Reply:
I assume that one doesn't because it doesn't need to: it's being used from ValueType.Equals/GetHashCode, so it was only being invoked in a situation where `IEquatable<T>` isn't relevant because it wouldn't have been used anyway (plus `IEquatable<T>` was added much later).